code/modules/automation/auto_grinder.dm

```diff
@@ -0,0 +1,78 @@
+//A test machine that grinds up a item if applicable and outputs all the results as a reagent patch
+/obj/machinery/automation/grinder
+	name = "autogrinder"
+	desc = "Grinds up items with reagents inside and outputs it as a patch."
+	var/obj/item/reagent_containers/output_container = new/obj/item/reagent_containers/pill/patch() //The typepath of the reagent container we want to output; patch by default, also allows pills
+	var/amount_to_transfer = 10 //How many units should be dispensed into the output container
+	var/name_of_output = "" //If we want to append a custom name to the output we will, otherwise just uses default setup AKA get_master_reagent and total volume
+
+	radial_categories = list(
+	"Change Output Name",
+	"Change Transfer Rate",
+	"Change Container Output"
+	)
+
+	var/list/valid_containers = list(
+	new/obj/item/reagent_containers/pill(),
+	new/obj/item/reagent_containers/pill/patch()
+	)
+
+/obj/machinery/automation/grinder/Initialize()
+	. = ..()
+	for(var/obj/item/reagent_containers/stuff in valid_containers)
+		valid_containers[stuff] = image(stuff)
+
+/obj/machinery/automation/grinder/examine(mob/user)
+	. = ..()
+	if(.)
+		to_chat(user, "<span class='notice'>Current outputted name: <span class='bold'>[name_of_output ? name_of_output : "Default naming scheme"]</span></span>")
+		to_chat(user, "<span class='notice'>Units of reagents transferred to new: <span class='bold'>[amount_to_transfer ? amount_to_transfer : "No output!!!"]</span></span>")
+
+/obj/machinery/automation/grinder/MakeRadial(mob/living/user)
+	var/category = show_radial_menu(user, src, radial_categories, null, require_near = TRUE)
+	if(category)
```

Comment:
early return. I'm going to stop repeating this just fix the lack oif early return everywhere